src/Compiler/pars.fsy

```diff
@@ -3572,7 +3584,19 @@ constrPattern:
         SynPat.LongIdent(lid, None, None, args, vis, m) }
 
   | COLON_QMARK atomTypeOrAnonRecdType %prec pat_isinst
-      { SynPat.IsInst($2, lhs parseState) }
+      { let m = unionRanges (rhs parseState 1) $2.Range
+        SynPat.IsInst($2, m) }
+
+  | COLON_QMARK recover %prec pat_isinst
+      { let mColon = rhs parseState 1
```

Comment:
Not `reportParseErrorAt` here?

Reply:
No, it's handled by `recover`.

Reply:
Thanks, I've somehow forgotten about it!